From be84fa8e2dc41652213cbe09c523c248fb4f9614 Mon Sep 17 00:00:00 2001
From: George Tsiamasiotis <george@tsiamasiotis.gr>
Date: Tue, 23 Sep 2025 23:11:19 +0300
Subject: [PATCH] scripts/patch-kernel.sh: do not try to check files after
 patch

Since we are not using patch -b, *.orig files are only created when
there are conflicts, or never according to posix patch.

As such, it doesn't really make sense to always delete *.orig files
presuming they are patch backups, even if they are patch backups.
Doing so is both deleting potentially useful information for failed
patch applications and creating hard to diagnose bugs [1].

In a similar vein, checking for *.rej files does not add any value
since we're already checking the patch command's return code.

[1]: https://github.com/openwrt/packages/issues/27485

Signed-off-by: George Tsiamasiotis <george@tsiamasiotis.gr>
---
 scripts/patch-kernel.sh | 9 ---------
 lang/rust/Makefile | 2 +-
 2 file changed, 1 insertion(+), 9 deletions(-)

diff --git a/scripts/patch-kernel.sh b/scripts/patch-kernel.sh
index 52750dde6edf19..84bcba8858cfc1 100755
--- a/scripts/patch-kernel.sh
+++ b/scripts/patch-kernel.sh
@@ -43,12 +43,3 @@ for i in ${patchdir}/${patchpattern} ; do
 	exit 1
     fi
 done
-
-# Check for rejects...
-if [ "`find $targetdir/ '(' -name '*.rej' -o -name '.*.rej' ')' -print`" ] ; then
-    echo "Aborting.  Reject files found."
-    exit 1
-fi
-
-# Remove backup files
-find $targetdir/ '(' -name '*.orig' -o -name '.*.orig' ')' -exec rm -f {} \;
diff --git a/feeds/packages/lang/rust/Makefile b/feeds/packages/lang/rust/Makefile
index 717b15d46e..220a14f2e2 100644
--- a/feeds/packages/lang/rust/Makefile
+++ b/feeds/packages/lang/rust/Makefile
@@ -75,7 +75,7 @@ HOST_CONFIGURE_ARGS = \
 	--release-channel=stable \
 	--enable-cargo-native-static \
 	--bootstrap-cache-path=$(DL_DIR)/rustc \
-	--set=llvm.download-ci-llvm=true \
+	--set=llvm.download-ci-llvm=false \
 	$(TARGET_CONFIGURE_ARGS)
 
 define Host/Uninstall
